ogcapi-features/README.md

```diff
@@ -157,12 +157,11 @@ The following Link relations must exist in the Item object returned from the `/c
 | **rel**  | **href**                                     | **Media Type**       | **From**            | **Description**                                     |
 | -------- | -------------------------------------------- | -------------------- | ------------------- | --------------------------------------------------- |
 | `root`   | `/`                                          | application/json     | STAC API - Features | The root URI                                        |
+| `collection` | `/collections/{collectionId}`            | application/json     | STAC Collection     | The containing Collection |
 | `parent` | `/collections/{collectionId}`                | application/json     | OAFeat              | Parent reference, usually the containing Collection |
 | `self`   | `/collections/{collectionId}/items/{itemId}` | application/geo+json | OAFeat              | Self reference                                      |
 
-Note that the `parent` link for an Item should point to the containing Collection
-(e.g., `/collections/{collectionId}`), rather than the API sub-path
-of `/collections/{collectionId}/items/`.
+The `parent` link for an item may point to a Collection or a Catalog.  A `collection` link for an Item will always point to the containing Collection.  Links to a Collection must point to the `/collections/{collectionId}` endpoint, rather than the API sub-path of `/collections/{collectionId}/items/`.
```

Comment:
```suggestion
The `parent` link for an Item may point to a Collection or a Catalog.  The `collection` link for an Item will always point to the containing Collection.  Links to a Collection must point to the `/collections/{collectionId}` endpoint, rather than the API sub-path of `/collections/{collectionId}/items`.
```

Reply:
Sorry, I was bit unclear about this in the call. The / at the end for non-files is usually only required in self links, but there it's very important. I would say it's also a best practice to follow with all other links, but it may not be so much of an issue there as usually you always resolve against self if present.

Here's the corresponding stac-spec PR: https://github.com/radiantearth/stac-spec/pull/1214 @philvarner

Reply:
References: 
- https://github.com/stac-utils/pystac-client/pull/373
- https://github.com/stac-utils/pystac-client/pull/405
- https://github.com/radiantearth/stac-browser/issues/241

Reply:
@m-mohr - I think this is probably better for a dedicated ticket, but it looks to me like you are talking about a problem with incorrect relative links rather than a problem with trailing slashes.

If the `self` link is `https://example.com/some/resource` and the URL for a related resource is `https://example.com/some/resource/related`, then a correct relative URL is `./resource/related`.

I feel like requiring that URLs end with a slash may be overdoing it.  And this will lead to inconsistencies with OGC API specs (OGC API – Features uses URLs like `collections/{collectionId}`, `collections/{collectionId}/items`, `collections/{collectionId}/items/{itemId}` etc.).

I know that it can be easy to get relative URLs wrong, and that resolving absolute URLs can be tricky.  But both can be done correctly without requiring that URLs end with a slash.

My personal opinion is that the OGC APIs and perhaps STAC are a bit **over**specified in requiring both specific links and specific URLs.  And I think things would be improved if OGC APIs allowed trailing slashes (it can be easier to stand up static sites that work with trailing slashes).  But I know it is too late for that.

Reply:
@tschaub Indeed, I just posted it here because we stumbled across it in this PR during the STAC community call.
Whether it's an invalid self link or relative link is not so important I think, the main issue is that it resolves incorrectly. But indeed we should maybe also update the stac-spec to mention that you can also solve this by changing the relative links. It's just simpler to add a slash then update 20 relative links in links and assets...

> My personal opinion is that the OGC APIs and perhaps STAC are a bit overspecified in requiring both specific links and specific URLs. And I think things would be improved if OGC APIs allowed trailing slashes (it can be easier to stand up static sites that work with trailing slashes). But I know it is too late for that.

I'm not sure... is it really specified in OGC APIs that trailing slashes are not allowed?

Reply:
> But indeed we should maybe also update the stac-spec to mention that you can also solve this by changing the relative links. It's just simpler to add a slash then update 20 relative links in links and assets

I think I'm missing context.  It sounds like you are starting with a problem that I don't know about.  Does the spec have language or examples in it that create this problem?

> is it really specified in OGC APIs that trailing slashes are not allowed?

No, the OGC specs do not forbid the use of trailing slashes.  But they do require that services respond (with a 200) to requests without trailing slashes.  For example, in the OGC API – Features spec, requirements 18 and 19 essentially say "requests to `/collections/{collectionId}` must respond with a 200 that includes collection metadata."  That doesn't mean that they cannot also respond to `/collections/{collectionId}/` (with a trailing slash) – but as you know the content of the response in those two cases needs to be different if (dot) relative URLs are used in links.

Again, I might not be seeing the problem that you are starting with, but I think if the STAC specs say anything about all this, it should be advice about being careful with (dot) relative URLs.  I don't see trailing slashes as magically solving all problems resolving relative URLs.  And I also don't see this as just a problem in resolving relative URLs against a "self" link.  Some implementations may use the "self" link as the base, but it also seems reasonable that a client might use a request URL as a base and ignore the "self" link when resolving relative URLs (assuming that links in the response will be relative to the request).  So I think that any best practice advice given in the spec should say that care needs to be taken to ensure that *all* links (self and other related links) are correctly formatted.

Reply:
=> https://github.com/radiantearth/stac-spec/pull/1217